includes/importer/class-newspack-listings-importer.php

```diff
@@ -381,7 +603,24 @@ function( $cat_name ) {
 		if ( self::$is_dry_run ) {
 			WP_CLI::success( $post['post_title'] . ' imported successfully.' );
 		} else {
-			$post_id = wp_insert_post( $post );
+			if ( $existing_post ) {
+				// If the post has already been imported, don't update the content as it may have already been worked on.
+				if ( 'content' === self::$skip ) {
+					unset( $post['post_content'] );
+				}
+				$post_id = wp_update_post( $post );
+			} else {
+				$post_id = wp_insert_post( $post );
+			}
+
+			// GDG only: we don't want Yoast primary categories.
```

Comment:
Does this mean that for different publishers, the code will have to be updated? If someone used this CSV importer, would they lose primary categories assignments?

Reply:
Correct—since GDG was the first and only publisher to use the importer, I built in code specific to their import process to the import script. I'm not sure how much will need to be altered for future publishers, but if it's a lot it may be worth moving this code to the CMS importer repo instead.

Reply:
That said, I tried to write the script in a way that any blocks of code marked `GDG only` can simply be commented out/deleted for a more "generic" import.